### PR TITLE
WIP: Modernize layer to use new primitives

### PR DESCRIPTION
**Pull request by marcoceppi**
This is much like battlemidget/juju-layer-nginx#6 but pushes even further on the idea. I made sure `rubylib` is available for backwards compat, but ultimately this will make things work as most others do today.

The last thing I want to do is build a custom tactic that helps warn/error for those using dependency.txt in their charm so that people can update without problem

### CODE REVIEW

**Comment by marcoceppi** [comment]
This is blocked on juju/charm-tools#252

File: tactics/ruby.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+from charmtools.build.tactics import ExactMatch, Tactic
+
+
+class DependenciesTxtTactic(ExactMatch, Tactic):
+    FILENAME = 'dependencies.txt'
+    kind = "dynamic"
+
+    def __call__(self):
+        raise Exception('dependencies.txt is deprecated see '
+                        'https://github.com/battlemidget/juju-layer-ruby/pull/8')
```
Comment (adam-stokes) [inline tactics/ruby.py:11]: this is very cool

**Comment by jamesbeedy** [comment]
but it does bad things! -> http://paste.ubuntu.com/23112942/

see https://github.com/battlemidget/juju-layer-ruby/issues/9

File: tactics/ruby.py (hunk 1 above)
Comment (marcoceppi) [reply tactics/ruby.py:11]: It'll be even cooler when it works

**Comment by adam-stokes** [comment]
@jamesbeedy still a WIP, I reverted the previous PR so you should be ok

**Comment by marcoceppi** [comment]
@jamesbeedy yeah, you're hitting juju/charm-tools#252

**Comment by adam-stokes** [comment]
whats the status on this?

**Comment by adam-stokes** [comment]
I'm going to remerge this and work to fix any issues

**Pull request merged**